File: LocalSearchOptimization/Parameters/CoreParameters.cs
using System.Collections.Generic;
using LocalSearchOptimization.Components;

namespace LocalSearchOptimization.Parameters
{
    public class CoreParameters
    {
        public string Name { get; set; } = "core";

        public int Seed { get; set; } = 0;

        public bool DetailedOutput { get; set; } = true;

        public List<Operator> Operators { get; set; }

        public CoreParameters()
        {

        }

        protected CoreParameters(CoreParameters copy)
        {
            Name = copy.Name;
            Seed = copy.Seed;
            DetailedOutput = copy.DetailedOutput;
            Operators = copy.Operators;
        }

        public virtual CoreParameters Clone()
        {
            return new CoreParameters(this);
        }
    }
}

File: java/src/localsearchoptimization/parameters/CoreParameters.java
package localsearchoptimization.parameters;

import localsearchoptimization.components.Operator;

public class CoreParameters {

    public String name = "core";

    public int seed = 0;

    public boolean isDetailedOutput = true;

    public Operator[] operators;

    public CoreParameters() {

    }

    protected CoreParameters(CoreParameters copy) {
        name = copy.name;
        seed = copy.seed;
        isDetailedOutput = copy.isDetailedOutput;
        operators = copy.operators;
    }

    public CoreParameters Clone() {
        return new CoreParameters(this);
    }
}